_docs/protocols/2017-01-01-application-level.md

```diff
@@ -51,6 +51,40 @@ module.
 Every message type should have an instance of the `Message` typeclass.
 Please see [Time-Warp-NT guide](/protocols/time-warp-nt/#messaging).
 
+## Message names
+
+All messages are given custom names, since using full type names would be excessive. Each name is concatenation of one or two encoded `UnsignedVarInt`s.
```

Comment:
Section above seems to clearly define why each message should have name (from code point of view), and link at section's end references to why names are needed